src/OrchardCore.Modules/OrchardCore.Queries/Deployment/QueryBasedContentDeploymentStepDriver.cs

```diff
@@ -0,0 +1,81 @@
+using System.Threading.Tasks;
+using OrchardCore.Queries.ViewModels;
+using OrchardCore.Deployment;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Mvc.ModelBinding;
+using OrchardCore.Queries.Sql;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace OrchardCore.Queries.Deployment
+{
+    public class QueryBasedContentDeploymentStepDriver : DisplayDriver<DeploymentStep, QueryBasedContentDeploymentStep>
+    {
+        private readonly IQueryManager _queryManager;
+        private readonly IStringLocalizer S;
+
+        public QueryBasedContentDeploymentStepDriver(IQueryManager queryManager,
+            IStringLocalizer<QueryBasedContentDeploymentStepDriver> stringLocalizer)
+        {
+            _queryManager = queryManager;
+            S = stringLocalizer;
+        }
+
+        public override IDisplayResult Display(QueryBasedContentDeploymentStep step)
+        {
+            return
+                Combine(
+                    View("QueryBasedContentDeploymentStep_Fields_Summary", step).Location("Summary", "Content"),
+                    View("QueryBasedContentDeploymentStep_Fields_Thumbnail", step).Location("Thumbnail", "Content")
+                );
+        }
+
+        public override IDisplayResult Edit(QueryBasedContentDeploymentStep step)
+        {
+            return Initialize<QueryBasedContentDeploymentStepViewModel>("QueryBasedContentDeploymentStep_Fields_Edit", model =>
+            {
+                model.QueryName = step.QueryName;
+                model.QueryParameters = step.QueryParameters;
+                model.ExportAsSetupRecipe = step.ExportAsSetupRecipe;
+            }).Location("Content");
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(QueryBasedContentDeploymentStep step, IUpdateModel updater)
+        {
+            var model = new QueryBasedContentDeploymentStepViewModel();
+
+            if (await updater.TryUpdateModelAsync(model, Prefix, x => x.QueryName, x => x.QueryParameters, x => x.ExportAsSetupRecipe)) {
```

Comment:
Use descriptive variable names.

Reply:
Still ;).

Reply:
I've checked in many other places, and it looks like everywhere I've found same variable names. That's why I've decided to go with same short variable names here, just to be consistent. But, I will change it to be more descriptive.

Reply:
looks like this code above is outdated,

![image](https://user-images.githubusercontent.com/6403130/182649951-711d285f-4bd5-4cce-bac3-09a603296d0a.png)

@Piedone  which variable is still not descriptive enough?

Reply:
`x`

Reply:
what would you suggest? I used same variable name as it was used in other steps.

Reply:
Well, what are those variables storing? :) Are the three `x` variables the same thing?

Reply:
I expect Zoltan refers to `x` which is make sense ;)

Reply:
> Are the three `x` variables the same thing?

exactly @Piedone , this **x** is the same thing, it is an array of strongly typed lambda expressions, here it is **IUpdateModel** interface where you can see there is a **params** keyword:

![image](https://user-images.githubusercontent.com/6403130/182777652-ca0239ee-2be3-4b22-9464-9f9ae20ac1fb.png)

Reply:
we did this in many places in Orchard:

![image](https://user-images.githubusercontent.com/6403130/182778618-fbdfcfd9-f59b-4880-bac5-28e3baf3182c.png)

That's why I wasn't sure what do you mean by **Use descriptive variable names** 😛

Reply:
In many places in Orchard, `x` doesn't tell you what's happening either ;).

Reply:
so, what would you suggest @Piedone?

x here is related to expression, which should be included, I will change it to whatever you will prefer :)

Reply:
`viewModel` on line 48, `model` here?